core/src/main/java/io/undertow/server/handlers/accesslog/DefaultAccessLogReceiver.java

```diff
@@ -309,11 +325,59 @@ public void rotate() {
         }
     }
 
+    @SuppressWarnings("static-access")
     @Override
     public void close() throws IOException {
-        closed = true;
-        if (stateUpdater.compareAndSet(this, 0, 1)) {
-            logWriteExecutor.execute(this);
+        this.closed = true;
```

Comment:
method must take into account that the state could change between one if and the other. So, there is a possibility that all stateUpdater checks return false.